File: flutter/frameworks/agilite_flutter_core/lib/src/crud/models/crud_list_request.dart
class CrudListRequest {
  final int currentPage;
  final int pageSize;
  final String? search;
  final Map<String, dynamic>? customFilters;
  final Map<String, dynamic>? dialogMoreFiltersValue;
  final int? groupIndex;

  CrudListRequest({
    required this.currentPage,
    required this.pageSize,
    this.search,
    this.customFilters,
    this.dialogMoreFiltersValue,
    this.groupIndex,
  });

  Map<String, dynamic> toJson() {
    return {
      'currentPage': currentPage,
      'pageSize': pageSize,
      'search': search,
      'customFilters': customFilters,
      'groupIndex': groupIndex,
      'dialogMoreFiltersValue': dialogMoreFiltersValue,
    };
  }
}

File: kotlin/frameworks/agilite_kotlin_boot/src/main/kotlin/info/agilite/boot/crud/CrudListRequest.kt
package info.agilite.boot.crud

data class CrudListRequest(
  val currentPage: Int,
  val pageSize: Int,
  val search: String?,
  val customFilters: Map<String, Any?>?,
  val dialogMoreFiltersValue: Map<String, Any?>?,
  var groupIndex: Int?,
)
